### ISSUE TITLE
Signage for custom fizzlers

### ISSUE THREAD

**Issue by Neshiro** [BEEmod/BEE2-items#3042]
Hello, dear developers.
I've made signage images to be used with the custom fizzlers in order to know more easily what each fizzler lets pass through. One is for signaling that objects are blocked, another is for signaling that the player is blocked and the last is for signaling that portals are blocked. Map designers would place these next to custom fizzlers.
[custom fizzlers signage.zip](https://github.com/BEEmod/BEE2-items/files/3356779/custom.fizzlers.signage.zip)

I think these new signs would be useful, as everyone doesn't know the effects of every custom fizzler in the BeeMod. (Please find a jpeg and a vtf version of each sign in the attached folder.)

**Comment by Konclan** [comment]
These are a good suggestion but I believe they need to be tweaked to better suit the style of signages

**Comment by LautaroL20** [comment]
Nice contribution, although it is very incomplete yet.

**Comment by Neshiro** [comment]
If that's so, then what can I do to make it better ? Or do I let you handle this matter as of now ?

**Comment by LautaroL20** [comment]
We do not have the abbility to edit those signages properly. It would be better if you take care of them and finish those by yourself.

**Comment by Neshiro** [comment]
Alright then. What else would you need to call it complete ? Is there some files missing, or the quality of the images isn't good enough ? What do you need in addition to that ?

**Comment by Konclan** [comment]
I mean I can make signages, I will take a crack and them sometime.

**Comment by Neshiro** [comment]
I made some progression on the matter. You can test this working version of my signages. Keep in mind that it's still not finished and lacks polishing.

[signage_custom_fizzlers.zip](https://github.com/BEEmod/BEE2-items/files/3365059/signage_custom_fizzlers.zip)

**Comment by Libbybapa** [comment]
I appreciate the signage options you've presented.  I would point out that the simple line through a player or cube does not seem specific enough.  The line through the player does not differentiate between blocking the player (compressed smoke, force deflection, and closed solid) or killing the player (death fizzler and laser field).  Similarly, the line through the cube does not differentiate between fizzling a cube (emancipation or matter inquisition),  or simply preventing it from passing (force deflection, physics repulsion, or closed solid).  It seems that if using the signs, making them more specific in that regard would be beneficial.

**Comment by Neshiro** [comment]
I could debate with you that when the player is killed, like on the official signage for the laser field, you have a little skullface on the signage's figure. However, I will take your view into consideration. In the meantime, do you have any better idea so as to better depict the message behind each signage I'm making ?

**Comment by Konclan** [comment]
I put my attempt at the signage on the discord's #development channel if you want to check them out

**Comment by Neshiro** [comment]
I saw your proposal. Your images, albeit better in terms of neatness, are quite similar to mine, in my opinion. That's entirely fine, of course, however I would have appreciated if you had credited me for the inspiration.

**Comment by LautaroL20** [comment]
What is your username at the Discord´s server?

**Comment by LautaroL20** [comment]
I meant, your profile name (sorry, my english isn´t my maternal language). Because some people from here sometimes uses a different name at the Beemod´s discord channel that difficult identify who is who.

**Comment by Neshiro** [comment]
My name on the Discord server is Poisoned Bunny.

**Comment by Neshiro** [comment]
My name on the Discord server is the same, Neshiro.

**Comment by Libbybapa** [comment]
Neshiro I looked at your items.  The VTF files were not exactly set up right.  Here's a before picture:

![20190720213233_1](https://user-images.githubusercontent.com/33527465/61587109-64d96d80-ab38-11e9-9395-c9603b7c4b19.jpg)

Here's how they look after I adjusted the visible layer, the layer mask, and the vtf alpha settings.  

![20190720222100_1](https://user-images.githubusercontent.com/33527465/61587307-c8fe3080-ab3c-11e9-9919-6bb401b6a657.jpg)

I placed the updated VTF files in this folder for easy download:  https://drive.google.com/open?id=1gcPPfkpaf5d84h8vVyAEEAme7i21Up8N

After replacing the VTF files, I have found that the images sometimes do not update in existing maps until you add them to a NEW map that has not yet been compiles.  The images will then update when existing maps that had the previous images are compiled.

**Comment by Neshiro** [comment]
The bug you have in your first screenshot is something I had too before I managed to fix it by saving each of the three signages' alpha masks as a vtf. As it's something I fixed in the last package I sent here, I don't see how you could still have it.
Either way, to update a chamber that has these signages, what I found is that you have to remove them, build the map, exit, place them again and rebuid. That way they get updated.

May I ask you what you changed exactly in the files ?

**Comment by Libbybapa** [comment]
I might have downloaded the first version you posted.  It was a little while before I got to using them.

I am using Gimp for image editing.  I'm not sure what you are using or how similar the apps are.  In Gimp you can add a 'layer mask' and edit that.  The washed out areas that should have been black were fixed by making them black in the visible layer.  I then went to the layer mask, made the border black so the border isn't washed out (that's an issue with most of the other signs also) and then changed the uniform background of the layer mask to a light gray rather than solid white.

**Comment by Neshiro** [comment]
Who do I have to ask for this to maybe be put in a future release ?

**Comment by TeamSpen210** [comment]
The main thing is that I'd want these to be automatic in some way, so players don't need to configure each sign manually. Perhaps the signage would generate 4 in a line when set to this?

**Comment by Libbybapa** [comment]
I have actually used the 'no-cube' and 'no-player' signs and I know someone else who has used them also to denote places where the player or cube should not go.  I probably won't personally use them (or any other signs) for telling the properties of the custom fizzlers.  I think those properties are best simply learned by the player.  Experienced players already know the properties and inexperienced players don't play my maps.  In each map description I include a description of the properties of each custom fizzler used in that map and I usually include a tutorial room that highlights those properties along with any other particularly unusual moves that impact the logic.  

All of that said, I think the 'no-cube' and 'no-player' signs are useful as is and I'm glad to have them even if I have no plans to use them for their intended purpose.  :)

**Comment by Lieutenant-L-T-Smash** [comment]
I like this signage and I think it would be useful. It being automatic is a nice-to-have, but it's not necessary to make good use of these signs. This issue hasn't been touched in nearly 2 years. Can we get this included in the next version?

**Comment by Neshiro** [comment]
> 
> 
> I like this signage and I think it would be useful. It being automatic is a nice-to-have, but it's not necessary to make good use of these signs. This issue hasn't been touched in nearly 2 years. Can we get this included in the next version?

I agree that placing this signage isn't that hard to do manually. I also hope it will get included one day, but I doubt it.
